### PR TITLE
Mining consensus to swap algo with enable window (see PR 20)

### PR DESCRIPTION
**Pull request by 8bitcoder**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -2946,13 +2947,31 @@ bool ContextualCheckBlockHeader(const CBlockHeader& block, CValidationState& sta
...
+    //   previous YESCRYPT block was mined or;
+    //   mining majority and fork height reached
+    bool bAlgoSwitch1 = false;
+    if (nHeight >= consensusParams.nAlgoSwitchBlock1)
+    {
+        bAlgoSwitch1 = IsAlgoSwitch1Active(pindexPrev, consensusParams);
+        if (!bAlgoSwitch1)
+            bAlgoSwitch1 =
+                    (block.nVersion > 3) &&
+                    IsSuperMajority(3, pindexPrev, consensusParams.nMajorityEnableAlgoSwitch1, consensusParams);
```
Comment (cryptapus) [inline src/main.cpp]: Should this be version 4 instead of 3?

Comment (8bitcoder) [inline src/main.cpp]: Yes

File: src/chainparams.cpp [hunk 2]
```diff
@@ -70,6 +70,8 @@ class CMainParams : public CChainParams {
         consensus.nBlockAlgoNormalisedWorkDecayStart2 = 932000; // block where weight decay starts
         consensus.nGeoAvgWork_Start = 1400000;
         consensus.nAlgoSwitchBlock1 = 100000000;
+        consensus.nMajorityEnableAlgoSwitch1 = 600;
+        consensus.nAlgoSwitch1EnableWindow = 2000;
```
Comment (cryptapus) [inline src/chainparams.cpp]: For the testnet parameters, I added:
`consensus.nMajorityEnableAlgoSwitch1 = 60;`
`consensus.nAlgoSwitch1EnableWindow = 200;`

Comment (8bitcoder) [inline src/chainparams.cpp]: Updated the testnet parameters

Also set AlgoSwitchBlock1 same as BlockSequentialAlgoRuleStart3 to let these fork at the same time.

**Pull request merged**